mle/agents/coder.py

```diff
@@ -49,19 +52,37 @@ def __init__(self, model, working_dir='.', console=None):
 
         Your can leverage your capabilities by using the specific functions listed below:
 
-        1. Creating project structures based on the user requirement using function `create_directory`.
-        2. Writing clean, efficient, and well-documented code using function `create_file` and `write_file`.
-        3. Exam the project to re-use the existing code snippets as much as possible, you may need to use
+        - Creating project structures based on the user requirement using function `create_directory`.
+        - Writing clean, efficient, and well-documented code using function `create_file` and `write_file`.
+        - Exam the project to re-use the existing code snippets as much as possible, you may need to use
          functions like `list_files`, `read_file` and `write_file`.
-        4. Writing the code into the file when creating new files, do not create empty files.
-        5. Use function `preview_csv_data` to preview the CSV data if the task include CSV data processing.
-        6. Decide whether the task requires execution and debugging before moving to the next or not.
-        7. Generate the commands to run and test the current task, and the dependencies list for this task.
-        8. You only write Python scripts, don't write Jupiter notebooks which require interactive execution.
+        - Writing the code into the file when creating new files, do not create empty files.
+        - Use function `preview_csv_data` to preview the CSV data if the task include CSV data processing.
+        - Decide whether the task requires execution and debugging before moving to the next or not.
+        - Generate the commands to run and test the current task, and the dependencies list for this task.
+        - You only write Python scripts, don't write Jupiter notebooks which require interactive execution.
         """
+
+        if single_file:
```

Comment:
can we add a `todo` here?

the 2 prompts here have many redundancies.

Reply:
Sure